runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/internal/AwsChunkedReader.kt

```diff
@@ -157,6 +162,34 @@ internal class AwsChunkedReader(
         return signedChunk
     }
 
+    /**
+     * Get an unsigned aws-chunked encoding of [data].
+     * If [data] is not set, read the next chunk from [delegate] and add hex-formatted chunk size to the front.
+     * Note that this function will suspend until the whole chunk has been read OR the channel is exhausted.
+     * The unsigned chunk structure is: `string(IntHexBase(chunk-size)) + \r\n + chunk-data + \r\n`
+     *
+     * @param data the data which will be encoded to aws-chunked. if not provided, will default to
+     * reading up to [CHUNK_SIZE_BYTES] from [delegate].
+     * @return a buffer containing the chunked data or null if no data is available (channel is closed)
+     */
+    private suspend fun getUnsignedChunk(data: SdkBuffer? = null): SdkBuffer? {
+        val bodyBuffer = data ?: stream.readChunk()
+        bodyBuffer ?: return null
+
+        val unsignedChunk = SdkBuffer()
+
+        // headers
+        unsignedChunk.apply {
+            writeUtf8(bodyBuffer.size.toString(16))
+            writeUtf8("\r\n")
+        }
+
+        // append the body
+        unsignedChunk.write(bodyBuffer.readByteArray())
```

Comment:
fix: Don't use `readByteArray`, when working with buffers you want to try and stick to methods that take advantage of the internals of the buffer as much as possible (e.g. anything taking an `SdkSource` or `SdkBuffer` for instance). Internally these types transfer data ownership when possible rather than copying but if you force `readByteArray()` it's may end up doing two copies (1 (maybe) to get the data out of `bodyBuffer` and another to copy it into `unsignedChunk`).

This can just be `unsignedChunk.writeAll(bodyBuffer)`